### ISSUE TITLE
Keycloak admin get_group_by_path raises exception by not found

### ISSUE THREAD

**Issue by gbaru** [marcospereirampj/python-keycloak#617]
if a groups is not existing or cannot be found, get_groups_by_path raises an exception.
I inspected the code for this and I was able to find a turn around introducing the possible return codes as acceptable result:
Module keycloak_admin.py, line 1413: 
Instead of:
   "return raise_error_from_response(data_raw, KeycloakGetError)"
I replaced with:
    "return raise_error_from_response(data_raw, KeycloakGetError, [200, 404] )"
    
It works, but I would like to know if there are better solutions.

**Comment by Galaxy102** [comment]
A better way would have been:
```python
try:
    kc_group = keycloak_admin.get_group_by_path(path="/group")
except KeycloakGetError as e:
    if e.response_code != 404:
        raise
    else:
        kc_group = None
```
At least that's what I've used for a long time. With the fix from #627 I now have to do
```python
kc_group = keycloak_admin.get_group_by_path(path="/group")
if ("error", "Group path does not exist") in kc_group.items():
    kc_group = None
```
which feels a bit wrong.

**Comment by becelot** [comment]
While updating, we just ran into this issue as well. It feels wrong, especially since all the other get calls still raise a KeycloakGetError, for example Users, Group Members, realms, etc.

Are all the other APIs also migrated to the new error handling 'layout'?

We essentially wrapped the get_group_by_path method into a wrapper function, which redetects this error and throws the old exception. Which works a lot better than having this mix of error handling across the codebase.
